test/remote/gateways/remote_card_connect_test.rb

```diff
@@ -0,0 +1,191 @@
+    assert_equal 'Insufficient funds', response.message
+  end
+
+  def test_successful_authorize_and_capture
+    auth = @gateway.authorize(@amount, @credit_card, @options)
+    assert_success auth
+
+    assert capture = @gateway.capture(@amount, auth.authorization)
+    assert_success capture
+    assert_includes ['Approval Queued for Capture', 'Approval Accepted'], capture.message
+  end
+
+  def test_failed_authorize
+    response = @gateway.authorize(@amount, @declined_card, @options)
+    assert_failure response
+    assert_equal 'Insufficient funds', response.message
+  end
+
+  def test_partial_capture
+    auth = @gateway.authorize(@amount, @credit_card, @options)
+    assert_success auth
+
+    assert capture = @gateway.capture(@amount - 1, auth.authorization)
+    assert_success capture
+  end
+
+  def test_failed_capture
+    response = @gateway.capture(@amount, @invalid_txn)
+    assert_failure response
+    assert_equal 'Txn not found', response.message
+  end
+
+  def test_successful_echeck_purchase
+    response = @gateway.purchase(@amount, @check, @options)
+    assert_equal 'Success', response.message
+    assert_success response
+  end
+
+  def test_failed_echeck_purchase
+    response = @gateway.purchase(@amount, check(routing_number: '23433'), @options)
+    assert_failure response
+    assert_equal 'Invalid card', response.message
+  end
+
+  def test_successful_refund
+    purchase = @gateway.purchase(@amount, @credit_card, @options)
+    assert_success purchase
+
+    assert refund = @gateway.refund(@amount, purchase.authorization)
+    assert_success refund
+    assert_equal 'Approval', refund.message
+  end
+
+  def test_partial_refund
+    purchase = @gateway.purchase(@amount, @credit_card, @options)
+    assert_success purchase
+
+    assert refund = @gateway.refund(@amount - 1, purchase.authorization)
+    assert_success refund
+  end
+
+  def test_failed_refund
+    response = @gateway.refund(@amount, @invalid_txn)
+    assert_failure response
+    assert_equal 'Txn not found', response.message
+  end
+
+  def test_successful_void
+    auth = @gateway.authorize(@amount, @credit_card, @options)
+    assert_success auth
+
+    assert void = @gateway.void(auth.authorization)
+    assert_success void
+    assert_equal 'Approval', void.message
+  end
+
+  def test_failed_void
+    response = @gateway.void(@invalid_txn)
+    assert_failure response
+  end
+
+  def test_successful_verify
+    response = @gateway.verify(@credit_card, @options)
+    assert_success response
+    assert_match %r{Approval}, response.message
+  end
+
+  def test_failed_verify
+    response = @gateway.verify(@declined_card, @options)
+    assert_failure response
+    assert_match %r{Insufficient funds}, response.message
+  end
+
+  def test_invalid_login
+    gateway = CardConnectGateway.new(username: '', password: '', merchant_id: '')
+    assert_raises(ActiveMerchant::ResponseError) do
+      gateway.purchase(@amount, @credit_card, @options)
```

Comment:
Other gateways I've seen _usually_ have returned `STANDARD_ERROR_CODE[:config_error]` for this instead of throwing an exception. (In fact, I'm not used to seeing AM throw exceptions outside an adapter once the gateway's set up, period. I'm only used to seeing that as part of instantiation, such as the `requires!` calls.)